embark.el

```diff
@@ -466,55 +471,63 @@ There are three kinds:
 
 (defun embark-target-active-region ()
   "Target the region if active."
+  ;; TODO consider returning a string
   (when (use-region-p) '(region . <region>)))
 
 (autoload 'dired-get-filename "dired")
 
 (defun embark-target-file-at-point ()
   "Target file at point.
-This function mostly relies on `ffap-file-at-point', with two exceptions:
-
-1. In `dired-mode', it uses `dired-get-filename' instead.
-
-2. In `emacs-lisp-mode', it only calls `ffap-file-at-point' if
-   point is in a string or comment, or if it is on symbol
-   preceded by `require' or `use-package'."
-  (when-let ((file (cond
-                    ((derived-mode-p 'dired-mode)
-                     (dired-get-filename t 'no-error-if-not-filep))
-                    ((derived-mode-p 'emacs-lisp-mode)
-                     (when (or (nth 3 (syntax-ppss)) ; in a string
-                               (nth 4 (syntax-ppss)) ; or comment
-                               (save-excursion
-                                 (unless (looking-at "\\_<")
-                                   (forward-symbol -1))
-                                 (forward-symbol -1)
-                                 (looking-at "use-package\\|require")))
```

Comment:
I'm a little conflicted about removing this guard in Emacs Lisp files, requiring `use-package` or `require`. On the one hand, now with multiple targets we can loosen the criteria to allow more matches, on the other hand it causes the following problem:

Consider a `(pcase ...)` expression or some occurrence of the `align` command. Those are both functions (well, a macro and a function) and libraries which ffap targets. I'd like the function case to have priority over the file case. But now consider a file name in a string or comment in an Elisp buffer. It's typically also a symbol, but there I want the file interpretation to have priority.

Reply:
My general sentiment is to give files a lower priority than symbols. I would be fine if symbol wins always maybe except in dired.

What about this? 

1. Introduce an high priority file finder which only triggers in dired an for names with a file extension or a slash
2. Introduce a low priority file finder which triggers on all files, except if the high priority finder triggered?
3. Put the symbol finder between these finders.

Reply:
That sounds reasonable. For repeated targets the `delete-dups` call you put in keeps the first occurrence, right?

Reply:
Yes, but I am now thinking about removing it again. The problem is that it may not be effective in this case, so it only covers up bugs. Could it happen that the high priority finder returns "embark.el" and the low priority finder returns "embark"?

Reply:
I think this works the way I want: 6682da8f497b05739e1c06e41f0e1a0cd1a406ec

Reply:
I'd be OK with removing the `delete-dups` call. The commit I made makes the high priority finder not return anything in the library-without-extension case, so it doesn't rely on duplicate removal.

Reply:
I'd vote for removing the `delete-dups` call on the grounds that it's probably better to tighten up the target finders than to rely on duplicates being removed.

Reply:
I removed `delete-dups`: 275f1eb28470f77cd4c5ac6b3128681794c62de2.

Reply:
I just tried the updated version. Seems to work great!